src/handson/Table.js

```diff
@@ -0,0 +1,351 @@
+                        if (value instanceof WidgetReference) {
+                            var retVal = Handsontable.renderers.BaseRenderer.apply(this, arguments);
+
+                            var width = Math.max(context._table.getColWidth(col) || 50, context.minWidgetWidth()) - 8;
+                            var height = Math.max(context._table.getRowHeight(row) || 16, context.minWidgetHeight());
+                            if (!value.get(context).domNode) {
+                                value.get(context).domNode = document.createElement('DIV');
+                                value.get(context).widget
+                                    .width(width)
+                                    .height(height)
+                                    .target(value.get(context).domNode)
+
+                                ;
+                            }
+                            Handsontable.Dom.empty(td);
+                            td.appendChild(value.get(context).domNode);
+                            return retVal;
+                        }
+
+                        var field = context.fields()[col];
+                        var textRender = context.renderHtmlDataCells() ? Handsontable.renderers.HtmlRenderer : 
+
+Handsontable.renderers.TextRenderer;
+                        switch (field.type()) {
+                            case "string":
+                                value = field.transform(value);
+                                textRender.call(this, instance, td, row, col, prop, value, cellProperties);
+                                break;
+                            case "number":
+                                Handsontable.renderers.NumericRenderer.call(this, instance, td, row, col, prop, value, 
+
+cellProperties);
+                                break;
+                            case "boolean":
+                                Handsontable.renderers.CheckboxRenderer.call(this, instance, td, row, col, prop, value, 
+
+cellProperties);
+                                break;
+                            case "time":
+                                break;
+                            default:
+                                value = field.transform(value);
+                                textRender.call(this, instance, td, row, col, prop, value, cellProperties);
+                                break;
+                        }
+
+                        if (context._selectionBag.isSelected({ _id: row })) {
+                            td.style.color = "white";
+                            td.style.background = "#f48a00";
+
+                        } else {
+                            td.style.color = null;
+                            td.style.background = null;
+                        }
+                        if (row === context.data().length-1) {
+                               cellProperties.readOnly = false; // make cell read-only if it is first row or the text reads 'readOnly'
```

Comment:
Indentation + invalid comment - this makes the last row "not" readonly, is that correct?